programs/voter-stake-registry/src/lib.rs

```diff
@@ -207,3 +219,37 @@ pub struct Stake<'info> {
     #[account(executable)]
     pub rewards_program: UncheckedAccount<'info>,
 }
+
+impl Stake<'_> {
+    pub fn verify_delegate_and_its_mining(
```

Comment:
Great, thanks. A question - why not implementing the check over the instance of &self? This way the call will be simplified to `ctx.accounts.verify_delegate_and_its_mining(deposit_entry)?;`

Reply:
Because I'm dumb. Good catch! Fixed.